linera-core/src/client/mod.rs

```diff
@@ -223,6 +223,41 @@ impl<P, S: Storage + Clone> Client<P, S> {
         }
     }
 
+    /// Returns a clone with a different set of tracked chains.
+    pub fn clone_with(
+        &self,
+        validator_node_provider: P,
+        name: impl Into<String>,
+        tracked_chains: impl IntoIterator<Item = ChainId>,
+        long_lived_services: bool,
```

Comment:
I think the set of input arguments for overriding is bit arbitrary (or rather dictated by the current need) and a more generic approach with the builder pattern would be better in long run but it's a lot of typing and can wait.